sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -1623,9 +1627,7 @@ class Dataset[T] private[sql](
    */
   @scala.annotation.varargs
   def selectExpr(exprs: String*): DataFrame = sparkSession.withActive {
-    select(exprs.map { expr =>
-      Column(sparkSession.sessionState.sqlParser.parseExpression(expr))
-    }: _*)
+    select(exprs.map(functions.expr): _*)
```

Comment:
Expression parsing will use this SparkSession's parser, so there is no need to explicity parse anymore.